src/components/table.ts

```diff
@@ -228,39 +228,42 @@ export module Component {
     private _determineGuarantees(offeredWidths: number[], offeredHeights: number[]): _LayoutAllocation {
       var requestedWidths  = _Util.Methods.createFilledArray(0, this._nCols);
       var requestedHeights = _Util.Methods.createFilledArray(0, this._nRows);
-      var layoutWantsWidth  = _Util.Methods.createFilledArray(false, this._nCols);
-      var layoutWantsHeight = _Util.Methods.createFilledArray(false, this._nRows);
+      var columnNeedsWidth  = _Util.Methods.createFilledArray(false, this._nCols);
+      var rowNeedsHeight = _Util.Methods.createFilledArray(false, this._nRows);
       this._rows.forEach((row: AbstractComponent[], rowIndex: number) => {
         row.forEach((component: AbstractComponent, colIndex: number) => {
           var spaceRequest: _SpaceRequest;
           if (component != null) {
             spaceRequest = component._requestedSpace(offeredWidths[colIndex], offeredHeights[rowIndex]);
           } else {
-            spaceRequest = {width: 0, height: 0, wantsWidth: false, wantsHeight: false};
+            spaceRequest = {width: 0, height: 0};
           }
 
+
           var allocatedWidth = Math.min(spaceRequest.width, offeredWidths[colIndex]);
           var allocatedHeight = Math.min(spaceRequest.height, offeredHeights[rowIndex]);
+          var componentNeedsWidth = spaceRequest.width > offeredWidths[colIndex];
+          var componentNeedsHeight = spaceRequest.height > offeredHeights[rowIndex];
 
-          requestedWidths [colIndex] = Math.max(requestedWidths [colIndex], allocatedWidth );
+          requestedWidths[colIndex] = Math.max(requestedWidths[colIndex], allocatedWidth);
           requestedHeights[rowIndex] = Math.max(requestedHeights[rowIndex], allocatedHeight);
-          layoutWantsWidth [colIndex] = layoutWantsWidth [colIndex] || spaceRequest.wantsWidth;
-          layoutWantsHeight[rowIndex] = layoutWantsHeight[rowIndex] || spaceRequest.wantsHeight;
+
+          columnNeedsWidth[colIndex] = columnNeedsWidth[colIndex] || componentNeedsWidth;
+          rowNeedsHeight[rowIndex] = rowNeedsHeight[rowIndex] || componentNeedsHeight;
```

Comment:
Let's do a bit of variable reorganization?

I'm thinking of
```typescript
var allocatedWidth...
var allocatedHeight...
requestedWidths[colIndex]...
requestedHeights[rowIndex]...
var componentNeedsWidth...
var componentNeedsHeight...
...rest of code
```

The main thing I wanted to point out was you created this componenNeedsWidth variable but it is used... 4 lines later.  I think it'll be less confusing to read if you place the variable more next to the line where it is used.